src/main/java/org/mifos/connector/slcb/camel/routes/transfer/SLCBBaseRoutes.java

```diff
@@ -51,18 +51,26 @@ public void configure() throws Exception {
                     paymentRequestDTO.setInstitutionCode(slcbConfig.institutionCode);
                     paymentRequestDTO.setPurpose(purpose);
                     paymentRequestDTO.setRequestDate(requestDate);
-                    paymentRequestDTO.setSourceAccount(slcbConfig.sourceAccount);
+                    if (transactionList != null && !transactionList.isEmpty() &&
```

Comment:
it would be good if all these checks moved to a function.  can these checks precisely determine whether the specs are newer/ older

Reply:
Will think about this.